### PR TITLE
Retrofit response issue resolved,RecyclerView  event Handler Gson Date formatter added,batch sql added

### PR DESCRIPTION
**Pull request by Aelafseged**
RecyclerView click and long click handler for detail display and edit/update
Date serialization for n/w, added with Gson date formatter.
Batch add onto local sqlite after fetching from n/w with onconflict method
Retrofit response issue resolved on empty body.

### CODE REVIEW

**Comment by Aelafseged** [comment]
what date format shud we use for date_created/date_deleted?
and the primary key product_id... does it need to be auto_increment or server generated or user initiated?
cheers ;-)

**Comment by Aelafseged** [comment]
Have added the AfroSellDeskClient with edited product server side code

File: AfroSellAndroidClient/app/src/main/java/com/erihackton/shopping/source/remote/AfroSellRemoteDataSource.java [hunk 1]
```diff
@@ -21,7 +23,7 @@
 
 public class AfroSellRemoteDataSource implements DataSource {
     public static final String TAG = "AfroSellRemote";
-    public static final String baseUrl = "http://192.168.43.114:8000";
+    public static final String baseUrl = "http://192.168.43.114:3300";
```
Comment (EriHackathon) [inline AfroSellAndroidClient/app/src/main/java/com/erihackton/shopping/source/remote/AfroSellRemoteDataSource.java:26]: Aelaf, why do we need to change this port I didn't see any update in the server side actually.

File: product/src/main/java/com/erihackton/shopping/exceptions/ProductExceptionController.java [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+import org.springframework.web.bind.annotation.ExceptionHandler;
+
+@ControllerAdvice
+public class ProductExceptionController {
+	
+	@ExceptionHandler(value=ProductNotFoundException.class)
+			
+    public ResponseEntity<Object> exception(ProductNotFoundException ex){
+		return new ResponseEntity<>("Product Not FOUND ",HttpStatus.NOT_FOUND);
+
```
Comment (EriHackathon) [inline product/src/main/java/com/erihackton/shopping/exceptions/ProductExceptionController.java:15]: Dawit, I can see you have created an Exception class, but I don't see it where you are using it.

File: AfroSellAndroidClient/app/src/main/java/com/erihackton/shopping/source/remote/AfroSellRemoteDataSource.java (hunk 1 above)
Comment (Aelafseged) [reply AfroSellAndroidClient/app/src/main/java/com/erihackton/shopping/source/remote/AfroSellRemoteDataSource.java:26]: initially i was using my node.js server as a mock server, which was at port 8000 as to test my android app... now i'm using product spring server it's at 3300... but i'm continously changing the ip address..  when i'm using a internet cafe router that has dynamic ip and my phone hotspot

**Pull request merged**